maintenance-mode.php

```diff
@@ -3,56 +3,154 @@
+ * @param array $links The original array with customizer links.
+ * @return array $links The updated array with customizer links.
  */
-function vip_maintenance_mode_admin_notice__constant_not_set() {
-	if ( !current_user_can( 'activate_plugins' ) ) {
-		return;
-	}
+function vip_maintenance_mode_settings_link( $links ) {
+	$admin_url     = admin_url( 'customize.php?autofocus[control]=vip_maintenance_mode_enable' );
+	$settings_link = sprintf( '<a href="%s">%s</a>', $admin_url, __( 'Settings', 'maintenance-mode' ) );
+	array_unshift( $links, $settings_link );
+
+	return $links;
+}
+add_filter( 'plugin_action_links_' . plugin_basename( __FILE__ ), 'vip_maintenance_mode_settings_link' );
 
-	$class = 'notice notice-warning';
-	$message = __( 'Maintenance Mode won\'t work until you set the VIP_MAINTENANCE_MODE constant to <code>true</code>.', 'maintenance-mode' );
-	printf( '<div class="%1$s"><p>%2$s</p></div>', esc_attr( $class ), wp_kses( $message, array( 'code' => array() ) ) );
+/**
+ * Checks to see if the maintenance mode is activated
+ * 
+ * @since 0.2.3
+ * @return bool Return is maintenance mode is activated.
+ */
+function vip_maintenance_mode_is_activated() {
+	return get_option( 'vip_maintenance_mode_enable' ) || defined( 'VIP_MAINTENANCE_MODE' ) || current_user_can_bypass_vip_maintenance_mode();
 }
 
 /**
  * Checks to see if the current user can bypass maintenance mode
+ * 
+ * @since 0.1.0
+ * @return bool Whether the current user has the given capability.
  */
 function current_user_can_bypass_vip_maintenance_mode() {
-	/**
-	 * Filters the required capability to avoid the redirect to the maintenance page.
-	 *
-	 * @since 0.1.0
-	 */
 	$required_capability = apply_filters( 'vip_maintenance_mode_required_cap', 'edit_posts' );
 	return current_user_can( $required_capability );
 }
 
+/**
+ * Displays a notice in the admin bar to indicate that maintenance mode is enabled
+ *
+ * Only displayed to users who don't see the maintenance page.
+ *
+ * @since 0.1.1
+ */
+function vip_maintenance_mode_admin_bar_menu() {
+	global $wp_admin_bar;
+
+	if ( ! vip_maintenance_mode_is_activated() ) {
+		return;
+	}
+
+	$wp_admin_bar->add_menu( array(
+		'id'     => 'maintenance-mode',
+		'parent' => 'top-secondary',
+		'title'  => apply_filters( 'vip_maintenance_mode_admin_bar_title', __( 'Under maintenance', 'maintenance-mode' ) ),
+		'meta'   => array( 'class' => 'mm-notice' ),
+	) );
+}
+add_action( 'admin_bar_menu', 'vip_maintenance_mode_admin_bar_menu', 8 );
+
+/**
+ * Styles the admin bar notice
+ *
+ * @since 0.1.1
+ */
+function vip_maintenance_mode_admin_scripts() {
+	$styles = '#wpadminbar .mm-notice .ab-item {
+		background: #BF616A !important;
+		color: #ECEFF4 !important;
+		font-weight:500;
+	}';
+	wp_add_inline_style( 'admin-bar', $styles );
+}
+add_action( 'wp_enqueue_scripts', 'vip_maintenance_mode_admin_scripts' );
+add_action( 'admin_enqueue_scripts', 'vip_maintenance_mode_admin_scripts' );
+
+/**
+ * Enhance customizer
```

Comment:
I wonder why this existing code is coming up in the diff then.

Could be worth moving these maintenance items (fixing existing DocBlocks, etc.) to a new PR, since most of them are going to be related to the new feature of enabling MM via the UI.